modules/ProtocolBrowser/src/client/ProtocolBrowser.coffee

```diff
@@ -442,6 +443,34 @@ class ProtocolBrowserController extends Backbone.View
 		$(".bv_protocolBaseControllerContainer").addClass("hide")
 		$(".bv_noMatchesFoundMessage").addClass("hide")
 
+	downloadSELFile: =>	
+		dataToPost =
+			protocolCode: @protocolController.model.escape('codeName')
+
+		
+		# send the request to get a .csv of the SEL template file 
+		$.ajax
+			type: 'POST'
+			url: "/api/getTemplateSELFile"
+			data: dataToPost
+			timeout: 6000000
+			dataType: 'json'
+			success: (response) =>
+				#Since we can't directly send the .csv file to download it...
+				#...we create a hidden link with the download path and automatically click on it
+				# exporting and downloading the file to the user
+				encodedUri = encodeURI(response)
+				a = document.createElement('a');
+				a.style.display = 'none';
+				a.href = encodedUri;
+				a.setAttribute('target', '_blank')
+				document.body.appendChild(a);
+				a.click();
+			error: (err) =>
+				console.log "getTemplateSELFile() error:" + err
```

Comment:
Can you add something for the user to know that their download failed including the err message?